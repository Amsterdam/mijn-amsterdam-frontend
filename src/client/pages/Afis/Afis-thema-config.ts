import { ReactNode } from 'react';
import { AppRoutes } from '../../../universal/config/routes';
import { ZaakDetail } from '../../../universal/types';
import { MAX_TABLE_ROWS_ON_THEMA_PAGINA } from '../../config/app';
import {
  AfisFacturenByStateResponse,
  AfisFactuur,
  AfisFactuurState,
} from '../../../server/services/afis/afis-types';
import { DisplayProps } from '../../components/Table/TableV2';

export type AfisEmandateStub = ZaakDetail & Record<string, string>;

export type AfisFactuurFrontend = AfisFactuur & {
  factuurNummerEl: ReactNode;
};

export type AfisFacturenByStateFrontend = {
  [key in keyof AfisFacturenByStateResponse]: AfisFactuurFrontend[];
};

// Themapagina
const MAX_TABLE_ROWS_ON_THEMA_PAGINA_OPEN = 5;
const MAX_TABLE_ROWS_ON_THEMA_PAGINA_TRANSFERRED = 5;
const MAX_TABLE_ROWS_ON_THEMA_PAGINA_CLOSED = MAX_TABLE_ROWS_ON_THEMA_PAGINA;

const displayPropsFacturen: DisplayProps<AfisFactuurFrontend> = {
  afzender: 'Afzender',
  factuurNummerEl: 'Factuurnummer',
  statusDescription: 'Status',
  paymentDueDateFormatted: 'Vervaldatum',
};

export const listPageParamState: {
  [key in AfisFactuurState]: AfisFactuurState;
} = {
  open: 'open',
  closed: 'closed',
  transferred: 'transferred',
};

export type ListPageParamStateKey = keyof typeof listPageParamState;
export type ListPageParamState =
  (typeof listPageParamState)[ListPageParamStateKey];

export const listPageTitle = {
  [listPageParamState.open]: 'Openstaande facturen',
  [listPageParamState.closed]: 'Afgehandelde facturen',
  [listPageParamState.transferred]: 'Overgedragen facturen',
} as const;

export const facturenTableConfig = {
  [listPageParamState.open]: {
    title: listPageTitle[listPageParamState.open],
    subTitle:
      'De betaalstatus kan 3 werkdagen achterlopen op de doorgevoerde wijzigingen.',
    displayProps: displayPropsFacturen,
    maxItems: MAX_TABLE_ROWS_ON_THEMA_PAGINA_OPEN,
    listPageLinkLabel: 'Alle openstaande facturen',
  },
  [listPageParamState.closed]: {
    title: listPageTitle[listPageParamState.closed],
    displayProps: displayPropsFacturen,
    maxItems: MAX_TABLE_ROWS_ON_THEMA_PAGINA_CLOSED,
    listPageLinkLabel: 'Alle afgehandelde facturen',
  },
  [listPageParamState.transferred]: {
    title: listPageTitle[listPageParamState.transferred],
    displayProps: displayPropsFacturen,
    maxItems: MAX_TABLE_ROWS_ON_THEMA_PAGINA_TRANSFERRED,
    listPageLinkLabel: 'Alle afgehandelde facturen',
  },
} as const;

// Betaalvoorkeuren
const displayPropsEmandates: DisplayProps<AfisEmandateStub> = {
  name: 'Naam',
};

export const businessPartnerDetailsLabels = {
  fullName: 'Debiteurnaam',
  businessPartnerId: 'Debiteurnummer',
  email: 'E-mailadres factuur',
  phone: 'Telefoonnummer',
};

export const eMandateTableConfig = {
  active: {
    title: `Actieve automatische incasso's`,
    filter: (emandate: AfisEmandateStub) => emandate.isActive,
    displayProps: displayPropsEmandates,
  },
  inactive: {
    title: `Niet actieve automatische incasso's`,
    filter: (emandate: AfisEmandateStub) => !emandate.isActive,
    displayProps: displayPropsEmandates,
  },
} as const;

export const routes = {
  listPageFacturen: AppRoutes['AFIS/FACTUREN'],
  betaalVoorkeuren: AppRoutes['AFIS/BETAALVOORKEUREN'],
  themaPage: AppRoutes.AFIS,
} as const;
