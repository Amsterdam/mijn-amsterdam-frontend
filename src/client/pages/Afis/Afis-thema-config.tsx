import { ReactNode } from 'react';

import { generatePath, LinkProps } from 'react-router-dom';

import {
  AfisBusinessPartnerDetailsTransformed,
  AfisFacturenResponse,
  AfisFactuur,
  AfisFactuurState,
} from '../../../server/services/afis/afis-types';
import { AppRoutes } from '../../../universal/config/routes';
import { ZaakDetail } from '../../../universal/types';
import { DisplayProps } from '../../components/Table/TableV2';
import { MAX_TABLE_ROWS_ON_THEMA_PAGINA } from '../../config/app';

// Themapagina
const MAX_TABLE_ROWS_ON_THEMA_PAGINA_OPEN = 5;
const MAX_TABLE_ROWS_ON_THEMA_PAGINA_TRANSFERRED =
  MAX_TABLE_ROWS_ON_THEMA_PAGINA;
const MAX_TABLE_ROWS_ON_THEMA_PAGINA_CLOSED = MAX_TABLE_ROWS_ON_THEMA_PAGINA;

const displayPropsFacturenOpen: DisplayProps<AfisFactuurFrontend> = {
  afzender: 'Afzender',
  factuurNummerEl: 'Factuurnummer',
  paymentDueDateFormatted: 'Vervaldatum',
  statusDescription: 'Status',
};

const displayPropsFacturenAfgehandeld: DisplayProps<AfisFactuurFrontend> = {
  afzender: 'Afzender',
  factuurNummerEl: 'Factuurnummer',
  statusDescription: 'Status',
};

const displayPropsFacturenOvergedragen: DisplayProps<AfisFactuurFrontend> = {
  afzender: 'Afzender',
  factuurNummerEl: 'Factuurnummer',
  statusDescription: 'Status',
};

export const listPageTitle: Record<AfisFactuurState, string> = {
  open: 'Openstaande facturen',
  afgehandeld: 'Afgehandelde facturen',
  overgedragen:
    'Facturen in het incasso- en invorderingstraject van directie Belastingen',
};

export type AfisEmandateStub = ZaakDetail & Record<string, string>;

export type AfisFactuurFrontend = AfisFactuur & {
  factuurNummerEl: ReactNode;
};

export type AfisFacturenResponseFrontend = AfisFacturenResponse & {
  facturen: AfisFactuurFrontend[];
};

export type AfisFacturenByStateFrontend = {
  [key in AfisFactuurState]?: AfisFacturenResponseFrontend;
};

type AfisFacturenTableConfig = {
  title: string;
  subTitle?: ReactNode;
  displayProps: DisplayProps<AfisFactuurFrontend>;
  maxItems: number;
  listPageLinkLabel: string;
  listPageRoute: string;
  className: string;
};

type AfisFacturenTableConfigByState = Record<
  AfisFactuurState,
  AfisFacturenTableConfig
>;

export const facturenTableConfig: AfisFacturenTableConfigByState = {
  open: {
    title: listPageTitle.open,
    displayProps: displayPropsFacturenOpen,
    maxItems: MAX_TABLE_ROWS_ON_THEMA_PAGINA_OPEN,
    listPageLinkLabel: 'Alle openstaande facturen',
    listPageRoute: generatePath(AppRoutes['AFIS/FACTUREN'], {
      state: 'open',
    }),
    className: 'FacturenTable--open',
  },
  overgedragen: {
    title: listPageTitle.overgedragen,
    displayProps: displayPropsFacturenOvergedragen,
    maxItems: MAX_TABLE_ROWS_ON_THEMA_PAGINA_TRANSFERRED,
    listPageLinkLabel:
      'Alle facturen in het incasso- en invorderingstraject van directie Belastingen',
    listPageRoute: generatePath(AppRoutes['AFIS/FACTUREN'], {
      state: 'overgedragen',
    }),
    className: 'FacturenTable--afgehandeld',
  },
  afgehandeld: {
    title: listPageTitle.afgehandeld,
    displayProps: displayPropsFacturenAfgehandeld,
    maxItems: MAX_TABLE_ROWS_ON_THEMA_PAGINA_CLOSED,
    listPageLinkLabel: 'Alle afgehandelde facturen',
    listPageRoute: generatePath(AppRoutes['AFIS/FACTUREN'], {
      state: 'afgehandeld',
    }),
    className: 'FacturenTable--overgedragen',
  },
} as const;

// Betaalvoorkeuren
const displayPropsEmandates: DisplayProps<AfisEmandateStub> = {
  name: 'Naam',
};

export const businessPartnerDetailsLabels: DisplayProps<AfisBusinessPartnerDetailsTransformed> =
  {
    fullName: 'Debiteurnaam',
    businessPartnerId: 'Debiteurnummer',
    email: 'E-mailadres factuur',
    phone: 'Telefoonnummer',
    address: 'Adres',
  };

export const eMandateTableConfig = {
  active: {
    title: `Actieve automatische incasso's`,
    filter: (emandate: AfisEmandateStub) => emandate.isActive,
    displayProps: displayPropsEmandates,
  },
  inactive: {
    title: `Niet actieve automatische incasso's`,
    filter: (emandate: AfisEmandateStub) => !emandate.isActive,
    displayProps: displayPropsEmandates,
  },
} as const;

export const routes = {
  listPageFacturen: AppRoutes['AFIS/FACTUREN'],
  betaalVoorkeuren: AppRoutes['AFIS/BETAALVOORKEUREN'],
  themaPage: AppRoutes.AFIS,
} as const;

export const linkListItems: LinkProps[] = [
  {
    to: 'https://www.amsterdam.nl/ondernemen/afis/facturen/',
    title: 'Meer over facturen van de gemeente',
  },
  {
    to: import.meta.env.REACT_APP_SSO_URL_BELASTINGEN,
    title: 'Belastingen op Mijn Amsterdam',
  },
];
