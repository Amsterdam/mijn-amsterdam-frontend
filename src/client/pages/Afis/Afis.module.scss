@import '../../styles/_global.scss';

.FacturenTable {
  tr {
    th,
    td {
      &:nth-child(2) {
        white-space: normal;
        word-wrap: break-word;
      }
    }
  }
  @media screen and (min-width: $ams-breakpoint-medium) {
    &--afgehandeld,
    &--overgedragen {
      tr {
        th,
        td {
          &:nth-child(1),
          &:nth-child(2) {
            width: 25%;
          }
        }
      }
    }
  }
  &--open {
    tr {
      th,
      td {
        @media screen and (max-width: $ams-breakpoint-medium) {
          &:first-of-type {
            display: none;
          }
          &:nth-child(2) {
            padding-inline-start: 0 !important;
          }
        }
        @media screen and (min-width: $ams-breakpoint-medium) {
          width: 25%;
        }
      }
    }
  }
}
