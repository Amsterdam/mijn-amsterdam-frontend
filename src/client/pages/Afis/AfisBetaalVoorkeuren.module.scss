@import '../../styles/global';

.Datalist--businesspartnerdetails {
  white-space: pre-line;
}
