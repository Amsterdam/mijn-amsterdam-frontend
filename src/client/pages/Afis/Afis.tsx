import React from 'react';

import {
  Alert,
  Heading,
  Link,
  Paragraph,
  UnorderedList,
} from '@amsterdam/design-system-react';

import { AfisFactuurFrontend } from './Afis-thema-config';
import styles from './Afis.module.scss';
import { useAfisThemaData } from './useAfisThemaData.hook';
import { entries } from '../../../universal/helpers/utils';
import { MaButtonRouterLink } from '../../components/MaLink/MaLink';
import { PageContentCell } from '../../components/Page/Page';
import { ThemaTitles } from '../../config/thema';
import ThemaPagina from '../ThemaPagina/ThemaPagina';
import ThemaPaginaTable from '../ThemaPagina/ThemaPaginaTable';

function PageContentTop({
  urlNaarBelastingen,
}: {
  urlNaarBelastingen: string;
}) {
  return (
    <PageContentCell>
      <Paragraph className="ams-mb--sm">
        Hieronder ziet u een overzicht van uw facturen. Mist u een factuur of
        heeft u een vraag over één van uw facturen? Stuur een e-mail naar{' '}
        <Link href="mailto:debiteurenadministratie@amsterdam.nl?subject=Vraag over facturen en betaalvoorkeuren">
          debiteurenadministratie@amsterdam.nl
        </Link>{' '}
        met de details van de factuur, zoals het factuurnummer of andere
        relevante informatie, zodat zij u verder kunnen helpen.
      </Paragraph>
      <Paragraph>
        U ziet hier niet de facturen over Gemeentebelastingen. Deze vindt u
        terug bij{' '}
        <Link rel="noreferrer" href={urlNaarBelastingen}>
          Mijn Belastingen
        </Link>
        .
      </Paragraph>
    </PageContentCell>
  );
}

export function AfisDisclaimer() {
  return (
    <Alert severity="warning">
      <UnorderedList>
        <UnorderedList.Item>
          Het verwerken van uw betaling kan tot 4 werkdagen duren.
        </UnorderedList.Item>
        <UnorderedList.Item>
          Enkel het openstaande bedrag wordt getoond.
        </UnorderedList.Item>
        <UnorderedList.Item>
          Gebruik de betaallink alleen voor het voldoen van het volledige
          factuurbedrag. Indien er sprake is van een betalingsregeling of
          deelbetaling verzoeken we u het resterend bedrag handmatig over te
          maken onder vermelding van de gegevens op uw factuur.
        </UnorderedList.Item>
      </UnorderedList>
    </Alert>
  );
}

export function AfisDisclaimerOvergedragenFacturen() {
  return (
    <Alert>
      <Paragraph>
        Als u niet betaalt, wordt uw factuur door Financiën overgedragen naar de
        afdeling Incasso & Invordering van directie Belastingen. Deze afdeling
        is vanaf dat moment verantwoordelijk voor de invordering van uw factuur
        en daarmee uw aanspreekpunt. De status van uw factuur vindt u terug bij
        Mijn Belastingen - gemeente Amsterdam.
      </Paragraph>
      <Heading level={4} size="level-5">
        Heeft u vragen?
      </Heading>
      <Paragraph>
        Afdeling Incasso & Invordering is van maandag tot en met vrijdag tussen
        08.00 en 18.00 uur bereikbaar op{' '}
        <Link rel="noreferrer" href="tel:0202554800">
          020 255 4800
        </Link>
        . U kunt ook een e-mail sturen naar{' '}
        <Link rel="noreferrer" href="mailto:belastingen@amsterdam.nl">
          belastingen@amsterdam.nl
        </Link>
        . Noem in het onderwerp uw vorderingsnummer en team Incasso.
      </Paragraph>
    </Alert>
  );
}

export function AfisThemaPagina() {
  const {
    dependencyErrors,
    facturenByState,
    facturenTableConfig,
    isThemaPaginaError,
    isThemaPaginaLoading,
    listPageTitle,
    routes,
    linkListItems,
    belastingenLinkListItem,
  } = useAfisThemaData();

  const isPartialError = entries(dependencyErrors).some(
    ([, hasError]) => hasError
  );

  const pageContentSecondary = (
    <PageContentCell spanWide={12}>
      <MaButtonRouterLink
        className="ams-mb--sm"
        variant="secondary"
        href={routes.betaalVoorkeuren}
      >
        Betaalvoorkeuren
      </MaButtonRouterLink>
      <AfisDisclaimer />
    </PageContentCell>
  );

  const pageContentErrorAlert = (
    <>
      We kunnen niet alle gegevens tonen.{' '}
      {entries(dependencyErrors)
        .filter(([, hasError]) => hasError)
        .map(([state]) => (
          <React.Fragment key={state}>
            <br />- {listPageTitle[state]} kunnen nu niet getoond worden.
          </React.Fragment>
        ))}
    </>
  );

  const pageContentTables = entries(facturenTableConfig).map(
    ([
      state,
      {
        title,
        subTitle,
        displayProps,
        maxItems,
        listPageLinkLabel,
        listPageRoute,
        className,
      },
    ]) => {
      const subTitleNode =
        state === 'overgedragen' && !!facturenByState?.[state]?.facturen.length
          ? state === 'overgedragen' && <AfisDisclaimerOvergedragenFacturen />
          : subTitle;
      return (
        <ThemaPaginaTable<AfisFactuurFrontend>
          key={state}
          title={title}
          subTitle={subTitleNode}
          zaken={facturenByState?.[state]?.facturen ?? []}
          displayProps={displayProps}
          maxItems={maxItems}
          totalItems={facturenByState?.[state]?.count}
          listPageLinkLabel={listPageLinkLabel}
          listPageRoute={listPageRoute}
          className={styles[className]}
        />
      );
    }
  );

  return (
    <ThemaPagina
      title={ThemaTitles.AFIS}
      isError={isThemaPaginaError}
      isPartialError={isPartialError}
      errorAlertContent={pageContentErrorAlert}
      isLoading={!isThemaPaginaError && isThemaPaginaLoading}
      linkListItems={linkListItems}
      pageContentTop={
        <PageContentTop urlNaarBelastingen={belastingenLinkListItem.to} />
      }
      pageContentMain={
        <>
          {pageContentSecondary}
          {pageContentTables}
        </>
      }
    />
  );
}
