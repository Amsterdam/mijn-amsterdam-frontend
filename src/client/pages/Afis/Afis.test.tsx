import { render, waitFor } from '@testing-library/react';

import { generatePath } from 'react-router-dom';
import { MutableSnapshot } from 'recoil';
import { AppRoutes } from '../../../universal/config/routes';
import { AppState } from '../../../universal/types';
import { appStateAtom } from '../../hooks/useAppState';
import MockApp from '../MockApp';
import { AfisThemaPagina } from './Afis';
import { bffApi } from '../../../test-utils';
import {
  AfisBusinessPartnerDetailsTransformed,
  AfisFactuur,
} from '../../../server/services/afis/afis-types';

const businessPartnerIdEncrypted = 'xxx-123-xxx';
const testState = {
  AFIS: {
    status: 'OK',
    content: {
      isKnown: true,
      businessPartnerIdEncrypted,
    },
  },
} as AppState;

function initializeState(snapshot: MutableSnapshot) {
  snapshot.set(appStateAtom, testState);
}

describe('<Afis />', () => {
  const businessPartnerDetails: AfisBusinessPartnerDetailsTransformed = {
    addressId: 999,
    businessPartnerId: 515177,
    fullName: 'Taxon Expeditions BV',
    phone: null,
    email: null,
  };

  const mockFacturen = {
    open: [
      {
        id: 'F001',
        title: 'Invoice F001',
        afzender: 'Company A',
        datePublished: '2023-01-15',
        datePublishedFormatted: '15 januari 2023',
        paymentDueDate: '2023-02-15',
        paymentDueDateFormatted: '15 februari 2023',
        debtClearingDate: null,
        debtClearingDateFormatted: null,
        amountOwed: 1000,
        amountOwedFormatted: '€ 1.000,00',
        factuurNummer: 'F001',
        status: 'openstaand',
        paylink: 'https://payment.example.com/F001',
        documentDownloadLink: 'https://download.example.com/F001',
      },
      {
        id: 'F003',
        title: 'Invoice F003',
        afzender: 'Company C',
        datePublished: '2023-03-10',
        datePublishedFormatted: '10 maart 2023',
        paymentDueDate: '2023-04-10',
        paymentDueDateFormatted: '10 april 2023',
        debtClearingDate: null,
        debtClearingDateFormatted: null,
        amountOwed: 2000,
        amountOwedFormatted: '€ 2.000,00',
        factuurNummer: 'F003',
        status: 'openstaand',
        paylink: 'https://payment.example.com/F003',
        documentDownloadLink: 'https://download.example.com/F003',
      },
    ],
    closed: [
      {
        id: 'F002',
        title: 'Invoice F002',
        afzender: 'Company B',
        datePublished: '2023-02-01',
        datePublishedFormatted: '1 februari 2023',
        paymentDueDate: '2023-03-01',
        paymentDueDateFormatted: '1 maart 2023',
        debtClearingDate: '2023-02-15',
        debtClearingDateFormatted: '15 februari 2023',
        amountOwed: 0,
        amountOwedFormatted: '€ 0,00',
        factuurNummer: 'F002',
        status: 'betaald',
        paylink: null,
        documentDownloadLink: 'https://download.example.com/F002',
      },
    ],
  };

  bffApi
    .get(`/services/afis/businesspartner/${businessPartnerIdEncrypted}`)
    .reply(200, {
      content: businessPartnerDetails,
      status: 'OK',
    });
  bffApi
    .get(`/services/afis/facturen/overzicht/${businessPartnerIdEncrypted}`)
    .reply(200, {
      content: mockFacturen,
      status: 'OK',
    });

  const routeEntry = generatePath(AppRoutes.AFIS);
  const routePath = AppRoutes.AFIS;

  const Component = () => (
    <MockApp
      routeEntry={routeEntry}
      routePath={routePath}
      component={AfisThemaPagina}
      initializeState={initializeState}
    />
  );

  it('Matches the Full Page snapshot', async () => {
    const screen = render(<Component />);
    expect(screen.asFragment()).toMatchSnapshot();

    await waitFor(() => {
      expect(screen.getByText('Company B')).toBeInTheDocument();
    });
  });
});
