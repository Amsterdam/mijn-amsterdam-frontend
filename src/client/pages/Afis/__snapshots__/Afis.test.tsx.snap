// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Afis /> > Matches the Full Page snapshot 1`] = `
<DocumentFragment>
  <div
    class="ams-grid"
  >
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-6-medium ams-grid__cell--span-7-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide _PageHeadingWrap_c5a3f1"
    >
      <div
        class="_PageHeadingInner_c5a3f1"
        id="skip-to-id-AppContent"
      >
        <nav
          class="ams-breadcrumb"
        >
          <ol
            class="ams-breadcrumb__list"
          >
            <li
              class="ams-breadcrumb__item"
            >
              <a
                class="ams-breadcrumb__link"
                href="/"
              >
                Home
              </a>
            </li>
            <li
              class="ams-breadcrumb__item"
            >
              <a
                class="ams-breadcrumb__link"
                href="/facturen-en-betalen"
                tabindex="-1"
              >
                <span
                  class="ams-visually-hidden"
                >
                  Facturen en betalen
                </span>
              </a>
            </li>
          </ol>
        </nav>
        <h3
          class="ams-heading ams-heading--level-1 _PageHeading_c5a3f1"
        >
          Facturen en betalen
        </h3>
      </div>
    </div>
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-6-medium ams-grid__cell--span-6-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide"
    >
      <p
        class="ams-paragraph ams-mb--sm"
      >
        Hieronder ziet u een overzicht van uw facturen. Mist u een factuur of heeft u een vraag over één van uw facturen? Stuur een e-mail naar 
        <a
          class="ams-link ams-link--standalone"
          href="mailto:debiteurenadministratie@amsterdam.nl?subject=Vraag over facturen en betaalvoorkeuren"
        >
          debiteurenadministratie@amsterdam.nl
        </a>
         met de details van de factuur, zoals het factuurnummer of andere relevante informatie, zodat zij u verder kunnen helpen.
      </p>
      <p
        class="ams-paragraph"
      >
        U ziet hier niet de facturen over Gemeentebelastingen. Deze vindt u terug bij 
        <a
          class="ams-link ams-link--standalone"
          href="http://remote-api-host/sso/portaal/belastingen"
          rel="noreferrer"
        >
          Mijn Belastingen
        </a>
        .
      </p>
    </div>
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-6-medium ams-grid__cell--span-12-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide"
    >
      <ul
        class="ams-link-list"
      >
        <li>
          <a
            class="ams-link-list__link"
            href="https://www.amsterdam.nl/veelgevraagd/facturen-van-de-gemeente-controleren-gegevens-wijzigen-automatische-incasso-regelen-38caa"
            rel="noreferrer"
          >
            <span
              class="ams-icon ams-icon--size-5"
            >
              <svg
                aria-hidden="true"
                focusable="false"
                viewBox="0 0 32 32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m9.757 32-2.9-2.91L19.937 16 6.857 2.91 9.757 0l16 16z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
            Meer over betalen aan de gemeente
          </a>
        </li>
        <li>
          <a
            class="ams-link-list__link"
            href="http://remote-api-host/sso/portaal/belastingen"
            rel="noreferrer"
          >
            <span
              class="ams-icon ams-icon--size-5"
            >
              <svg
                aria-hidden="true"
                focusable="false"
                viewBox="0 0 32 32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m9.757 32-2.9-2.91L19.937 16 6.857 2.91 9.757 0l16 16z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
            Belastingen op Mijn Amsterdam
          </a>
        </li>
      </ul>
    </div>
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-6-medium ams-grid__cell--span-12-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide"
    >
      <a
        class="_MaButtonLink_aa758a ams-button ams-button--secondary ams-mb--sm"
        href="/facturen-en-betalen/betaalvoorkeuren"
      >
        Betaalvoorkeuren
      </a>
      <section
        class="ams-alert ams-alert--warning"
      >
        <div
          class="ams-alert__icon"
        >
          <span
            class="ams-icon ams-icon--size-4"
          >
            <svg
              aria-hidden="true"
              focusable="false"
              viewBox="0 0 36 36"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M33.63 30.9H2.37L18 3zm-24.44-4h17.62L18 11.19z"
              />
              <path
                d="M17 16.99h2v4.95h-2zm0 5.95h2v1.95h-2z"
              />
            </svg>
          </span>
        </div>
        <div
          class="ams-alert__content"
        >
          <h2
            class="ams-heading ams-heading--level-4"
          >
            Belangrijk om te weten
          </h2>
          <ul
            class="ams-unordered-list ams-unordered-list--no-markers"
          >
            <li
              class="ams-unordered-list__item"
            >
              Het verwerken van uw betaling kan tot 4 werkdagen duren.
            </li>
            <li
              class="ams-unordered-list__item"
            >
              Enkel het openstaande bedrag wordt getoond.
            </li>
            <li
              class="ams-unordered-list__item"
            >
              Gebruik de betaallink alleen voor het voldoen van het volledige factuurbedrag. Indien er sprake is van een betalingsregeling of deelbetaling verzoeken we u het resterend bedrag handmatig over te maken onder vermelding van de gegevens op uw factuur.
            </li>
          </ul>
        </div>
      </section>
    </div>
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-6-medium ams-grid__cell--span-12-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide"
    >
      <h3
        class="ams-heading ams-heading--level-2"
      >
        Openstaande facturen
      </h3>
      <div
        class="ams-table"
      >
        <table
          class="ams-table__table _TableV2_4c6825 _FacturenTable--open_1fc87f"
        >
          <thead
            class="ams-table__header"
          >
            <tr
              class="ams-table__row"
            >
              <th
                class="ams-table__header-cell"
              >
                Factuurnummer
              </th>
              <th
                class="ams-table__header-cell"
              >
                Afzender
              </th>
              <th
                class="ams-table__header-cell"
              >
                Vervaldatum
              </th>
              <th
                class="ams-table__header-cell"
              >
                Status
              </th>
            </tr>
          </thead>
          <tbody
            class="ams-table__body"
          >
            <tr
              class="ams-table__row"
            >
              <td
                class="ams-table__cell"
              >
                <span
                  class="_DocumentLinkWrap_81c3b4"
                >
                  <a
                    class="ams-link ams-link--standalone _DocumentLink_81c3b4 _MaRouterLink__no-default-underline_aa758a"
                    href="https://download.example.com/F001"
                  >
                    <span
                      class="_DocumentLinkIcon_81c3b4"
                    >
                      <span
                        class="ams-icon ams-icon--size-5"
                      >
                        <svg
                          aria-hidden="true"
                          focusable="false"
                          viewBox="0 0 32 32"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M0 27.996h32V32H0zm22.36-13.983 2.83 2.823L16 26.044l-9.19-9.208 2.83-2.823L14 18.377V0h4v18.377z"
                          />
                        </svg>
                      </span>
                    </span>
                    factuur F001
                  </a>
                </span>
              </td>
              <td
                class="ams-table__cell"
              >
                Company A
              </td>
              <td
                class="ams-table__cell"
              >
                15 februari 2023
              </td>
              <td
                class="ams-table__cell"
              >
                Openstaand: 
                <a
                  class="ams-link ams-link--standalone _MaRouterLink__no-underline-fat_aa758a"
                  href="https://payment.example.com/F001"
                  target="_blank"
                >
                  openstaand
                </a>
              </td>
            </tr>
            <tr
              class="ams-table__row"
            >
              <td
                class="ams-table__cell"
              >
                <span
                  class="_DocumentLinkWrap_81c3b4"
                >
                  <a
                    class="ams-link ams-link--standalone _DocumentLink_81c3b4 _MaRouterLink__no-default-underline_aa758a"
                    href="https://download.example.com/F003"
                  >
                    <span
                      class="_DocumentLinkIcon_81c3b4"
                    >
                      <span
                        class="ams-icon ams-icon--size-5"
                      >
                        <svg
                          aria-hidden="true"
                          focusable="false"
                          viewBox="0 0 32 32"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M0 27.996h32V32H0zm22.36-13.983 2.83 2.823L16 26.044l-9.19-9.208 2.83-2.823L14 18.377V0h4v18.377z"
                          />
                        </svg>
                      </span>
                    </span>
                    factuur F003
                  </a>
                </span>
              </td>
              <td
                class="ams-table__cell"
              >
                Company C
              </td>
              <td
                class="ams-table__cell"
              >
                10 april 2023
              </td>
              <td
                class="ams-table__cell"
              >
                Openstaand: 
                <a
                  class="ams-link ams-link--standalone _MaRouterLink__no-underline-fat_aa758a"
                  href="https://payment.example.com/F003"
                  target="_blank"
                >
                  openstaand
                </a>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-6-medium ams-grid__cell--span-12-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide"
    >
      <h3
        class="ams-heading ams-heading--level-2"
      >
        Facturen in het incasso- en invorderingstraject van directie Belastingen
      </h3>
      <div
        class="ams-table"
      >
        <table
          class="ams-table__table _TableV2_4c6825 _FacturenTable--afgehandeld_1fc87f"
        >
          <tbody
            class="ams-table__body"
          />
        </table>
      </div>
      <p
        class="ams-paragraph"
      >
        U heeft (nog) geen facturen in het incasso- en invorderingstraject van directie belastingen
      </p>
    </div>
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-6-medium ams-grid__cell--span-12-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide"
    >
      <h3
        class="ams-heading ams-heading--level-2"
      >
        Afgehandelde facturen
      </h3>
      <div
        class="ams-table"
      >
        <table
          class="ams-table__table _TableV2_4c6825 _FacturenTable--overgedragen_1fc87f"
        >
          <thead
            class="ams-table__header"
          >
            <tr
              class="ams-table__row"
            >
              <th
                class="ams-table__header-cell"
              >
                Factuurnummer
              </th>
              <th
                class="ams-table__header-cell"
              >
                Afzender
              </th>
              <th
                class="ams-table__header-cell"
              >
                Status
              </th>
            </tr>
          </thead>
          <tbody
            class="ams-table__body"
          >
            <tr
              class="ams-table__row"
            >
              <td
                class="ams-table__cell"
              >
                <span
                  class="_DocumentLinkWrap_81c3b4"
                >
                  <a
                    class="ams-link ams-link--standalone _DocumentLink_81c3b4 _MaRouterLink__no-default-underline_aa758a"
                    href="https://download.example.com/F002"
                  >
                    <span
                      class="_DocumentLinkIcon_81c3b4"
                    >
                      <span
                        class="ams-icon ams-icon--size-5"
                      >
                        <svg
                          aria-hidden="true"
                          focusable="false"
                          viewBox="0 0 32 32"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M0 27.996h32V32H0zm22.36-13.983 2.83 2.823L16 26.044l-9.19-9.208 2.83-2.823L14 18.377V0h4v18.377z"
                          />
                        </svg>
                      </span>
                    </span>
                    factuur F002
                  </a>
                </span>
              </td>
              <td
                class="ams-table__cell"
              >
                Company B
              </td>
              <td
                class="ams-table__cell"
              >
                betaald
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
